src/Range/RequestBodyRange.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace LaraCrafts\ChunkUploader\Range;
+
+use Illuminate\Http\Request;
+use InvalidArgumentException;
+
+class RequestBodyRange implements Range
+{
+    private $index;
+
+    private $numberOfChunks;
+
+    private $totalSize;
+
+    private $chunkSize;
+
+    /**
+     * RequestRange constructor.
+     *
+     * @param \Illuminate\Http\Request|\Symfony\Component\HttpFoundation\ParameterBag $request
+     * @param string $indexKey The index of the current chunk
+     * @param string $numberOfChunksKey The total number of the chunks
+     * @param string $chunkSizeKey The size of the chunk
+     * @param string $totalSizeKey The size of the original file
+     */
+    public function __construct($request, string $indexKey, string $numberOfChunksKey, string $chunkSizeKey, string $totalSizeKey)
+    {
+        if ($request instanceof Request) {
+            $request = $request->request;
+        }
+
+        $this->index = $request->get($indexKey);
+        $this->numberOfChunks = $request->get($numberOfChunksKey);
+        $this->chunkSize = $request->get($chunkSizeKey);
+        $this->totalSize = $request->get($totalSizeKey);
+
+        if ($this->numberOfChunks <= 0) {
```

Comment:
All of these errors seem kind of vague, so an error could be '-1 must be greater than zero', ok? But what is this -1? I think it would be better to throw an error like 'Number of chunks should be greater than 0, received -1'

Reply:
You are right. I will fix this.

Reply:
Actually the final errors will say `\`numberOfChunks\` must be greater than zero` as the parameter to `sprintf` is the name of the input and not the value it self.